# The Battle of the Neighborhoods - Week 2

### Import necessary libraries

In [2]:
import numpy as np 
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
import json 
!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim 
import requests 
from pandas.io.json import json_normalize 
import matplotlib.cm as cm
import matplotlib.colors as colors
from sklearn.cluster import KMeans
!conda install -c conda-forge folium=0.5.0 --yes 
import folium

print('Done! Libraries imported.')

Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2020.6.20  |       hecda079_0         145 KB  conda-forge
    openssl-1.1.1g             |       h516909a_1         2.1 MB  conda-forge
    geographiclib-1.50         |             py_0          34 KB  conda-forge
    geopy-2.0.0                |     pyh9f0ad1d_0          63 KB  conda-forge
    certifi-2020.6.20          |   py36h9f0ad1d_0         151 KB  conda-forge
    python_abi-3.6             |          1_cp36m           4 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         2.5 MB

The following NEW packages will be INSTALLED:

    geographiclib:   1.50-py_0          conda-forge
    geopy:           

## New York, New York

### Download datasets for NYC analysis and transform it into a pandas dataframe

In [3]:
#First we get the data table for the postal codes of NYC neighborhoods.
PostCodes_NY= pd.read_html('https://en.wikipedia.org/wiki/Neighborhoods_in_New_York_City', header=0)
df_NY= PostCodes_NY[0]
df_NY

Community Board(CB)  Areakm2  Pop.Census2010  Pop./km2  \
0           Bronx CB 1     7.17           91497     12761   
1           Bronx CB 2     5.54           52246      9792   
2           Bronx CB 3     4.07           79762     19598   
3           Bronx CB 4     5.28          146441     27735   
4           Bronx CB 5     3.55          128200     36145   
5           Bronx CB 6     4.01           83268     20765   
6           Bronx CB 7     4.84          139286     28778   
7           Bronx CB 8     8.83          101731     11521   
8           Bronx CB 9    12.41          172298     13884   
9          Bronx CB 10    16.76          120392      7183   
10         Bronx CB 11     9.32          113232     12149   
11         Bronx CB 12    14.56          152344     10463   
12       Brooklyn CB 1    12.82          160338     12507   
13       Brooklyn CB 2     7.72           98620     12775   
14       Brooklyn CB 3     7.67          143867     18757   
15       Brooklyn CB 4     5.31          104358     19653   
16       Brooklyn CB 5    14.61          173198     11855   
17       Brooklyn CB 6     9.01          104054     11549   
18       Brooklyn CB 7    10.96          120063     10955   
19       Brooklyn CB 8     4.25           96076     22606   
20       Brooklyn CB 9     4.07          104014     25556   
21      Brooklyn CB 10    10.57          122542     11593   
22      Brooklyn CB 11    10.18          172129     16909   
23      Brooklyn CB 12     9.32          185046     19855   
24      Brooklyn CB 13     8.88          106120     11950   
25      Brooklyn CB 14     7.61          168806     22182   
26      Brooklyn CB 15    12.82          160319     12505   
27      Brooklyn CB 16     4.97           85343     17172   
28      Brooklyn CB 17     8.73          165753     18987   
29      Brooklyn CB 18    24.68          194653      7887   
30      Manhattan CB 1     4.45           34420      7735   
31      Manhattan CB 2     4.01           93119     23222   
32      Manhattan CB 3     4.56          164407     36054   
33      Manhattan CB 4     5.41           87479     16170   
34      Manhattan CB 5     4.25           44028     10360   
35      Manhattan CB 6     3.55          136152     38353   
36      Manhattan CB 7     5.46          207699     38040   
37      Manhattan CB 8     5.13          217063     42312   
38      Manhattan CB 9     3.91          111724     28574   
39     Manhattan CB 10     3.63          107109     29507   
40     Manhattan CB 11     5.75          117743     20477   
41     Manhattan CB 12     7.64          208414     27279   
42         Queens CB 1    14.97          211220     14110   
43         Queens CB 2    13.36          109920      8228   
44         Queens CB 3     7.33          169083     23067   
45         Queens CB 4     6.16          167005     27111   
46         Queens CB 5    19.63          165911      8452   
47         Queens CB 6     7.59          115967     15279   
48         Queens CB 7    32.97          242952      7369   
49         Queens CB 8    19.17          146594      7647   
50         Queens CB 9    10.08          141608     14048   
51        Queens CB 10    16.19          127274      7861   
52        Queens CB 11    24.99          116404      4658   
53        Queens CB 12    25.46          223602      8782   
54        Queens CB 13    33.31          196284      5893   
55        Queens CB 14    23.28          106686      4583   
56  Staten Island CB 1    36.62          162609      4440   
57  Staten Island CB 2    54.62          127071      2326   
58  Staten Island CB 3    58.97          152908      2593   
59       New York City   831.39         8008278      9632   

                                        Neighborhoods  
0                    Melrose, Mott Haven, Port Morris  
1                               Hunts Point, Longwood  
2   Claremont, Concourse Village, Crotona Park, Mo...  
3                               Concourse, Highbridge  
4

In [4]:
#Rename the columns
df_NY= df_NY.rename(columns= {'Community Board(CB)':'Borough', 'Areakm2':'Area (in km^2)', 'Pop.Census2010':'Population (2010)', 'Pop./km2':'Population (in km^2)'})
df_NY= df_NY.drop([59,1]) #I drop the general New York City row at the bottom of the dataframe
df_NY.head()

Borough  Area (in km^2)  Population (2010)  Population (in km^2)  \
0  Bronx CB 1            7.17              91497                 12761   
2  Bronx CB 3            4.07              79762                 19598   
3  Bronx CB 4            5.28             146441                 27735   
4  Bronx CB 5            3.55             128200                 36145   
5  Bronx CB 6            4.01              83268                 20765   

                                       Neighborhoods  
0                   Melrose, Mott Haven, Port Morris  
2  Claremont, Concourse Village, Crotona Park, Mo...  
3                              Concourse, Highbridge  
4  Fordham, Morris Heights, Mount Hope, Universit...  
5        Bathgate, Belmont, East Tremont, West Farms

In [5]:
Postal_Codes= ['10453', '10453', '10453', '10453', '10453', '10453', '10453', '10453', '10453', '10453', '10453', 
              '11212', '11212', '11212', '11212', '11212', '11212', '11212', '11212', '11212', '11212', '11212', 
              '11212', '11212', '11212', '11212', '11212', '11212', '11212', 
              '10026', '10026', '10026', '10026', '10026', '10026', '10026', '10026', '10026', '10026', '10026', 
              '10026', 
              '11361', '11361', '11361', '11361', '11361', '11361', '11361', '11361', '11361', '11361', '11361', 
              '11361', '11361', '11361', 
              '10302', '10302', '10302', 
              ]
df_NY['Postal Codes']= Postal_Codes
df_NY

Borough  Area (in km^2)  Population (2010)  \
0           Bronx CB 1            7.17              91497   
2           Bronx CB 3            4.07              79762   
3           Bronx CB 4            5.28             146441   
4           Bronx CB 5            3.55             128200   
5           Bronx CB 6            4.01              83268   
6           Bronx CB 7            4.84             139286   
7           Bronx CB 8            8.83             101731   
8           Bronx CB 9           12.41             172298   
9          Bronx CB 10           16.76             120392   
10         Bronx CB 11            9.32             113232   
11         Bronx CB 12           14.56             152344   
12       Brooklyn CB 1           12.82             160338   
13       Brooklyn CB 2            7.72              98620   
14       Brooklyn CB 3            7.67             143867   
15       Brooklyn CB 4            5.31             104358   
16       Brooklyn CB 5           14.61             173198   
17       Brooklyn CB 6            9.01             104054   
18       Brooklyn CB 7           10.96             120063   
19       Brooklyn CB 8            4.25              96076   
20       Brooklyn CB 9            4.07             104014   
21      Brooklyn CB 10           10.57             122542   
22      Brooklyn CB 11           10.18             172129   
23      Brooklyn CB 12            9.32             185046   
24      Brooklyn CB 13            8.88             106120   
25      Brooklyn CB 14            7.61             168806   
26      Brooklyn CB 15           12.82             160319   
27      Brooklyn CB 16            4.97              85343   
28      Brooklyn CB 17            8.73             165753   
29      Brooklyn CB 18           24.68             194653   
30      Manhattan CB 1            4.45              34420   
31      Manhattan CB 2            4.01              93119   
32      Manhattan CB 3            4.56             164407   
33      Manhattan CB 4            5.41              87479   
34      Manhattan CB 5            4.25              44028   
35      Manhattan CB 6            3.55             136152   
36      Manhattan CB 7            5.46             207699   
37      Manhattan CB 8            5.13             217063   
38      Manhattan CB 9            3.91             111724   
39     Manhattan CB 10            3.63             107109   
40     Manhattan CB 11            5.75             117743   
41     Manhattan CB 12            7.64             208414   
42         Queens CB 1           14.97             211220   
43         Queens CB 2           13.36             109920   
44         Queens CB 3            7.33             169083   
45         Queens CB 4            6.16             167005   
46         Queens CB 5           19.63             165911   
47         Queens CB 6            7.59             115967   
48         Queens CB 7           32.97             242952   
49         Queens CB 8           19.17             146594   
50         Queens CB 9           10.08             141608   
51        Queens CB 10           16.19             127274   
52        Queens CB 11           24.99             116404   
53        Queens CB 12           25.46             223602   
54        Queens CB 13           33.31             196284   
55        Queens CB 14           23.28             106686   
56  Staten Island CB 1           36.62             162609   
57  Staten Island CB 2           54.62             127071   
58  Staten Island CB 3           58.97             152908   

    Population (in km^2)                                      Neighborhoods  \
0                  12761                   Melrose, Mott Haven, Port Morris   
2                  19598  Claremont, Concourse Village, Crotona Park, Mo...   
3                  27735                              Concourse, Highbridge   
4                  36145  Fordham, Morris Heights, Mount Hope, Universit...   
5                  2

As you can see on the table above, there are 5 boroughs in New York: Bronx, Brooklyn, Manhattan, Queens, and Staten Island. 

__*NOTE*__: These postal codes were obtained from https://www.health.ny.gov/statistics/cancer/registry/appendix/neighborhoods.htm ; Only one postal code for each borough (from the central part of the borough) was recorded for simplicity and visualization purposes.

In [6]:
df_NY.shape

(58, 6)

### Get the sum of the population

*_Note:_* This population value is from 2010, but later on in this code you can see and compare with the current population (from census.gov).

In [7]:
total_pop= df_NY['Population (2010)'].sum()
total_pop

8006205

### Census data for average income, businesses, and retail sales

Obtain and read in data from the NY census to analyze in the report how population, businesses, and the economy of the city might influence the future of the clothing store. 

In [8]:
census_NY= pd.read_html('https://www.census.gov/quickfacts/fact/table/newyorkcitynewyork/PST040219#qf-flag-X')
census_NY

[  All Topics Population Age and Sex Race and Hispanic Origin Population Characteristics Housing Families & Living Arrangements Computer and Internet Use Education Health Economy Transportation Income & Poverty Businesses Geography  \
 0  Population estimates base, April 1, 2010, (V2019)                                                                                                                                                                                     
 
    New York city, New York  
 0                  8175031  ,
                                            Population Unnamed: 1
 0         Population estimates, July 1, 2019, (V2019)    8336817
 1   Population estimates base, April 1, 2010, (V2019)    8175031
 2   Population, percent change - April 1, 2010 (es...       2.0%
 3                   Population, Census, April 1, 2010    8175133
 4                                         Age and Sex        NaN
 5                      Persons under 5 years, percent     6.5%
 6   

As you can see above, the estimate for population from census.gov in 2010 is around 8175031 while our sum of the population by borough is a just a little less (around 8006205). The population from census.gov in 2019 is higher, at around 8336817. We can calculate the population growth rate to determine how many future clients there might be.

In [9]:
pop2010= 8175031
pop2019= 8336817
pop_growth= ((pop2019-pop2010)/(pop2010))*100
print('The population growth rate (from 2010-2019) was around:', pop_growth, '%')

The population growth rate (from 2010-2019) was around: 1.9790261345797957 %


In 9 years the population grew by almost 2%. This means we could approximate the population in 9 years to be around: 

In [10]:
futurepop= (pop2019*(pop_growth/100))+pop2019
futurepop

8501804.787222091

### Data for costs of living in NY

This information can tell us how much people need in order to eat, pay rent, commute, and spend on other necessities. If the income we see is much higher than what the average person needs, then they're more likely to spend more money on luxury items such as this clothing store.

In [11]:
livingNY= pd.read_html('https://www.numbeo.com/cost-of-living/in/New-York')
livingNY

[    0                                                  1
 0 NaN  whenDocReady(function() {  jQuery('#city_selec...,     0      1
 0 NaN  Tweet,                                           Restaurants     [ Edit ]  \
 0                        Meal, Inexpensive Restaurant      21.00 $   
 1   Meal for 2 People, Mid-range Restaurant, Three...     100.00 $   
 2      McMeal at McDonalds (or Equivalent Combo Meal)       9.25 $   
 3                      Domestic Beer (1 pint draught)       7.00 $   
 4                  Imported Beer (12 oz small bottle)       8.00 $   
 5                                Cappuccino (regular)       4.90 $   
 6                     Coke/Pepsi (12 oz small bottle)       2.19 $   
 7                          Water (12 oz small bottle)       1.77 $   
 8                                             Markets     [ Edit ]   
 9                          Milk (regular), (1 gallon)       4.42 $   
 10                   Loaf of Fresh White Bread (1 lb)       3.36 $   
 11 

### Create a map of NYC Boroughs

In [12]:
from IPython.display import display

In [13]:
latitudeNYC= 40.730610
longitudeNYC= -73.935242

latBronx= 40.8448
lonBronx= -73.8648
latBrook= 40.6782
lonBrook= -73.9442
latManh= 40.7831
lonManh= -73.9712
latQueens= 40.7282
lonQueens= -73.7949
latSI= 40.5795
lonSI= -74.1502

map_NY = folium.Map(location=[latitudeNYC, longitudeNYC], zoom_start=12)

folium.Marker([latBronx, lonBronx],
              popup='Bronx',
              icon=folium.Icon(color='green')
             ).add_to(map_NY)
folium.Marker([latBrook, lonBrook],
              popup='Brooklyn',
              icon=folium.Icon(color='red')
             ).add_to(map_NY)
folium.Marker([latManh, lonManh],
              popup='Manhattan',
              icon=folium.Icon(color='purple')
             ).add_to(map_NY)
folium.Marker([latQueens, lonQueens],
              popup='Queens',
              icon=folium.Icon(color='blue')
             ).add_to(map_NY)
folium.Marker([latSI, lonSI],
              popup='Staten Island',
              icon=folium.Icon(color='pink')
             ).add_to(map_NY)

map_NY

In [288]:
CLIENT_ID = '1IVZFEEO0D5XWSHQUVP0UIMMFSPSPDXWYFMCCFMFEMEAMLXJ' 
CLIENT_SECRET = 'LRRZSM13LBPVZ1PHJQX0PDZML2C1QJQM2HNQ2JEXSWAK4S2V' 
VERSION = '20180604'
LIMIT = 30

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)


Your credentails:
CLIENT_ID: 1IVZFEEO0D5XWSHQUVP0UIMMFSPSPDXWYFMCCFMFEMEAMLXJ
CLIENT_SECRET:LRRZSM13LBPVZ1PHJQX0PDZML2C1QJQM2HNQ2JEXSWAK4S2V


### Search on Foursquare for every popular clothing venue in each NY borough

In [369]:
address = 'New York, NY'
latitude= 40.7493
longitude= -73.9893
print(latitude, longitude)

40.7493 -73.9893


In [370]:
search_query = 'mall'
radius = 500
print(' Searching for:',search_query +  '.... OK!')

 Searching for: mall.... OK!


In [371]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=1IVZFEEO0D5XWSHQUVP0UIMMFSPSPDXWYFMCCFMFEMEAMLXJ&client_secret=LRRZSM13LBPVZ1PHJQX0PDZML2C1QJQM2HNQ2JEXSWAK4S2V&ll=40.7493,-73.9893&v=20180604&query=mall&radius=500&limit=30'

In [372]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5f371ebaccdcdc5e3f22532e'},
 'response': {'venues': [{'id': '4a19bdd6f964a520477a1fe3',
    'name': 'Manhattan Mall',
    'location': {'address': '100 W 33rd St',
     'crossStreet': 'b/t Avenue Of The Americas & West 33rd St',
     'lat': 40.749055139214676,
     'lng': -73.98895146714219,
     'labeledLatLngs': [{'label': 'entrance',
       'lat': 40.749391,
       'lng': -73.989106}],
     'distance': 40,
     'postalCode': '10001',
     'cc': 'US',
     'neighborhood': 'New York, NY',
     'city': 'New York',
     'state': 'NY',
     'country': 'United States',
     'formattedAddress': ['100 W 33rd St (b/t Avenue Of The Americas & West 33rd St)',
      'New York, NY 10001',
      'United States']},
    'categories': [{'id': '4bf58dd8d48988d1fd941735',
      'name': 'Shopping Mall',
      'pluralName': 'Shopping Malls',
      'shortName': 'Mall',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/shops/mall_',
       'suffix': '.png

In [373]:
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe

categories  hasPerk  \
0   [{'id': '4bf58dd8d48988d1fd941735', 'name': 'S...    False   
1   [{'id': '4bf58dd8d48988d178941735', 'name': 'D...    False   
2   [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...    False   
3   [{'id': '4bf58dd8d48988d108951735', 'name': 'W...    False   
4   [{'id': '4bf58dd8d48988d111951735', 'name': 'J...    False   
5   [{'id': '4bf58dd8d48988d103951735', 'name': 'C...    False   
6   [{'id': '4bf58dd8d48988d103951735', 'name': 'C...    False   
7   [{'id': '4bf58dd8d48988d164941735', 'name': 'P...    False   
8   [{'id': '4bf58dd8d48988d10b951735', 'name': 'V...    False   
9   [{'id': '4bf58dd8d48988d105951735', 'name': 'K...    False   
10  [{'id': '4bf58dd8d48988d1f6941735', 'name': 'D...    False   
11  [{'id': '4d954afda243a5684865b473', 'name': 'O...    False   
12  [{'id': '4bf58dd8d48988d10c951735', 'name': 'C...    False   
13  [{'id': '4bf58dd8d48988d124941735', 'name': 'O...    False   
14  [{'id': '4bf58dd8d48988d1fa931735', 'name': 'H...    False   

                          id                   location.address location.cc  \
0   4a19bdd6f964a520477a1fe3                      100 W 33rd St          US   
1   4bbf6366006dc9b6627ffc3f            901 Ave of the Americas          US   
2   4a802155f964a520a9f41fe3         875 Avenue of the Americas          US   
3   4bcc8c5d937ca5931349ab92                       901 6th Ave.          US   
4   4d921bb6f5388cfac7e8c43d                                NaN          US   
5   4bd5c40b5631c9b61ea4a430                        901 6th Ave          US   
6   4df10dd881303a516711eec6                     901 6th Avenue          US   
7   4da35b13b3e7236afbfc3079  Broadway btwn 39th & 42nd Streets          US   
8   4a6f3c21f964a520b0d51fe3              1282 Broadway Ste 249          US   
9   4c694548c946e21e06ade98e                      100 W 33rd St          US   
10  4a92fa2bf964a5205e1e20e3                      100 W 32nd St          US   
11  4c77e9e89f51a0931c8db627                      100 W 33rd St          US   
12  4ee691577ee50feb35639907            900 Ave of the Americas          US   
13  53badfbc498e61971d7dc43a                                NaN          US   
14  4b07764af964a520cffd22e3                      1155 Broadway          US   

   location.city location.country                       location.crossStreet  \
0       New York    United States  b/t Avenue Of The Americas & West 33rd St   
1       New York    United States                                  W 33rd St   
2       New York    United States                               at W 31st St   
3       New York    United States                              at W 32nd St.   
4            NaN    United States                                        NaN   
5       New York    United States                    btwn W 34th & W 33rd St   
6       New York    United States                                        NaN   
7       New York    United States                                        NaN   
8       New York    United States                               at W 33rd St   
9       New York    United States                             Manhattan Mall   
10      New York    United States                                 at 6th Ave   
11      New York    United States                                        NaN   
12      New York    United States                             Manhattan Mall   
13      New York    United States                                        NaN   
14      New York    United States                                    27th St   

    location.distance                          location.formattedAddress  \
0                  40  [100 W 33rd St (b/t Avenue Of The Americas & W...   
1                 118  [901 Ave of the Americas (W 33rd St), New York...   
2                   3  [875 Avenue of the Americas (at W 31st St), Ne...   
3                   4  [901 6th Ave. (at W 32nd St.), New York, NY 10...   
4                 234                          [New York, Un

In [374]:
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]

dataframe_filtered

name        categories  \
0                                  Manhattan Mall     Shopping Mall   
1                           Manhattan Mall Dental  Dentist's Office   
2                                       Starbucks       Coffee Shop   
3                                      Strawberry     Women's Store   
4                             Golden Mall Jewelry     Jewelry Store   
5                                         Express    Clothing Store   
6                                           Virtu    Clothing Store   
7   Broadway Pedestrian Mall - 39th St to 42nd St             Plaza   
8                                        GameStop  Video Game Store   
9                           P.S. from Aéropostale        Kids Store   
10                                       JCPenney  Department Store   
11                                   LensCrafters      Optical Shop   
12                                        SEPHORA    Cosmetics Shop   
13                               Mallory & Church            Office   
14                           Broadway Plaza Hotel             Hotel   

                              address  cc      city        country  \
0                       100 W 33rd St  US  New York  United States   
1             901 Ave of the Americas  US  New York  United States   
2          875 Avenue of the Americas  US  New York  United States   
3                        901 6th Ave.  US  New York  United States   
4                                 NaN  US       NaN  United States   
5                         901 6th Ave  US  New York  United States   
6                      901 6th Avenue  US  New York  United States   
7   Broadway btwn 39th & 42nd Streets  US  New York  United States   
8               1282 Broadway Ste 249  US  New York  United States   
9                       100 W 33rd St  US  New York  United States   
10                      100 W 32nd St  US  New York  United States   
11                      100 W 33rd St  US  New York  United States   
12            900 Ave of the Americas  US  New York  United States   
13                                NaN  US  New York  United States   
14                      1155 Broadway  US  New York  United States   

                                  crossStreet  distance  \
0   b/t Avenue Of The Americas & West 33rd St        40   
1                                   W 33rd St       118   
2                                at W 31st St         3   
3                               at W 32nd St.         4   
4                                         NaN       234   
5                     btwn W 34th & W 33rd St        18   
6                                         NaN        26   
7                                         NaN       520   
8                                at W 33rd St        95   
9                              Manhattan Mall        52   
10                                 at 6th Ave        70   
11                                        NaN        92   
12                             Manhattan Mall       105   
13                                        NaN       247   
14                                    27th St       541   

                                     formattedAddress  \
0   [100 W 33rd St (b/t Avenue Of The Americas & W...   
1   [901 Ave of the Americas (W 33rd St), New York...   
2   [875 Avenue of the Americas (at W 31st St), Ne...   
3   [901 6th Ave. (at W 32nd St.), New York, NY 10...   
4                           [New York, United States]   
5   [901 6th Ave (btwn W 34th & W 33rd St), New Yo...   
6   [901 6th Avenue, New York, NY 10001, United St...   
7   [Broadway btwn 39th & 42nd Streets, New York, ...   
8   [1282 Broadway Ste 249 (at W 33rd St), New Yor...   
9   [100 W 33rd St (Manhattan Mall), New York, NY ...   
10  [100 W 32nd St (at 6th Ave), New York, NY 1000...   
11  [100 W 33rd St, New York, NY 10001, United Sta...   
12  [900 Ave of the Americas (Manhattan Mall), New...   
13                      [New York, NY, Unite

In [380]:
dataframe_filtered.name
dataframe_filtered= dataframe_filtered[0:10]

As can be seen from the above table, this isn't the most accurate representation of clothing stores in NYC. You can use this same code to search for specific stores (i.e. Zara, lululemon, Ted Baker, Nike, etc.) or even department stores (i.e. Macy's, Nordstrom, Bloomingdales, Saks Fifth Avenue, etc.). The owners of the store could search for stores that they know have around the same price range as theirs. I specifically used coordinates closer to Manhattan mall to to avoid overcrowding of data.

### Generate map of the New York venues found

In [381]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=15)

# add a red circle marker to represent New York City
folium.features.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='New York, New York',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

#for lat, lng, label in zip(dataframe_filtered.lat, dataframe_filtered.lng, dataframe_filtered.categories):
#    folium.features.CircleMarker(
#        [lat, lng],
#        radius=5,
#        color='blue',
#        popup=label,
#        fill = True,
#        fill_color='blue',
#        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map

## Miami, Florida

Now that we have analyzed NY, we move on to analyzing Miami, Florida. Then in the report, I will compare both based on the data analyzed here. 

### Download datasets for Miami analysis and transform it into a pandas dataframe

In [143]:
PostCodes_Mia= pd.read_html('https://en.wikipedia.org/wiki/List_of_neighborhoods_in_Miami', header=0)
df_Mia= PostCodes_Mia[0]
df_Mia

Neighborhood          Demonym            Population2010  \
0                      Allapattah              NaN                     54289   
1   Arts & Entertainment District              NaN                     11033   
2                        Brickell      Brickellite                     31759   
3                     Buena Vista              NaN                      9058   
4                   Coconut Grove          Grovite                     20076   
5                       Coral Way              NaN                     35062   
6                 Design District              NaN                      3573   
7                        Downtown       Downtowner  71,000 (13,635 CBD only)   
8                       Edgewater              NaN                     15005   
9                         Flagami              NaN                     50834   
10              Grapeland Heights              NaN                     14004   
11                Health District              NaN                      2705   
12                   Liberty City              NaN                     19725   
13                   Little Haiti              NaN                     29760   
14                  Little Havana              NaN                     76163   
15                    Lummus Park              NaN                      3027   
16                        Midtown        Midtowner                         -   
17                       Overtown           Towner                      6736   
18                      Park West              NaN                      4655   
19                      The Roads              NaN                      7327   
20                 Upper Eastside  Upper Eastsider                     12525   
21               Venetian Islands              NaN                       NaN   
22                   Virginia Key              NaN                        14   
23                   West Flagler              NaN                     31407   
24                        Wynwood       Wynwoodian                      7277   
25                          Miami          Miamian                    399457   

   Population/Km²                                  Sub-neighborhoods  \
0            4401                                                NaN   
1            7948                                                NaN   
2           14541                                      West Brickell   
3            3540  Buena Vista East Historic District and Design ...   
4            3091  Center Grove, Northeast Coconut Grove, Southwe...   
5            4496  Coral Gate, Golden Pines, Shenandoah, Historic...   
6            3623                                                NaN   
7           10613  Brickell, Central Business District (CBD), Dow...   
8            6675                                                NaN   
9            5665           Alameda, Grapeland Heights, and Fairlawn   
10           4130                                                NaN   
11           2148                                                NaN   
12           3733                                                NaN   
13           3840                      Lemon City (aka Little River)   
14           8423  Riverside and South River Drive Historic District   
15           3680                                                NaN   
16              -                              Edgewater and Wynwood   
17           3405                                      Spring Garden   
18           3635                                                NaN   
19           4899                                                NaN   
20           2513  Bay Point Estates, Bayside District, Belle Mea...   
21            NaN               Biscayne Island and San Marco Island   
22              -                                                NaN   
23           4428                                                NaN   
24           2983  Wynwood Art District and Wynwood Fashion 

In [141]:
df_Mia.shape

(26, 6)

__*NOTE:*__ Miami does not have "boroughs" as does NYC, so we use its neighborhoods for analysis purposes.

In [149]:
#Rename the columns
df_Mia= df_Mia.rename(columns= {'Population2010':'Population (2010)', 'Population/Km²': 'Population (per Km²)'})
df_Mia.head()

Neighborhood      Demonym Population (2010)  \
0                     Allapattah          NaN             54289   
1  Arts & Entertainment District          NaN             11033   
2                       Brickell  Brickellite             31759   
3                    Buena Vista          NaN              9058   
4                  Coconut Grove      Grovite             20076   

  Population (per Km²)                                  Sub-neighborhoods  \
0                 4401                                                NaN   
1                 7948                                                NaN   
2                14541                                      West Brickell   
3                 3540  Buena Vista East Historic District and Design ...   
4                 3091  Center Grove, Northeast Coconut Grove, Southwe...   

     Coordinates Postal Codes  
0  25.815-80.224        33125  
1  25.799-80.190        33132  
2  25.758-80.193        33131  
3  25.813-80.192        33137  
4  25.712-80.257        33133

In [150]:
Postal_CodesMia= ['33125', '33132', '33131', '33137', '33133', '33129', '33127', '33128', '33137', 
                  '33125', '33125', '33136', '33127', '33127', '33130', '33136', '33136', '33136', 
                 '33132', '33129', '33138', '33139', '33149', '33125', '33127', '33132']
df_Mia['Postal Codes']= Postal_CodesMia
df_Mia

Neighborhood          Demonym         Population (2010)  \
0                      Allapattah              NaN                     54289   
1   Arts & Entertainment District              NaN                     11033   
2                        Brickell      Brickellite                     31759   
3                     Buena Vista              NaN                      9058   
4                   Coconut Grove          Grovite                     20076   
5                       Coral Way              NaN                     35062   
6                 Design District              NaN                      3573   
7                        Downtown       Downtowner  71,000 (13,635 CBD only)   
8                       Edgewater              NaN                     15005   
9                         Flagami              NaN                     50834   
10              Grapeland Heights              NaN                     14004   
11                Health District              NaN                      2705   
12                   Liberty City              NaN                     19725   
13                   Little Haiti              NaN                     29760   
14                  Little Havana              NaN                     76163   
15                    Lummus Park              NaN                      3027   
16                        Midtown        Midtowner                         -   
17                       Overtown           Towner                      6736   
18                      Park West              NaN                      4655   
19                      The Roads              NaN                      7327   
20                 Upper Eastside  Upper Eastsider                     12525   
21               Venetian Islands              NaN                       NaN   
22                   Virginia Key              NaN                        14   
23                   West Flagler              NaN                     31407   
24                        Wynwood       Wynwoodian                      7277   
25                          Miami          Miamian                    399457   

   Population (per Km²)                                  Sub-neighborhoods  \
0                  4401                                                NaN   
1                  7948                                                NaN   
2                 14541                                      West Brickell   
3                  3540  Buena Vista East Historic District and Design ...   
4                  3091  Center Grove, Northeast Coconut Grove, Southwe...   
5                  4496  Coral Gate, Golden Pines, Shenandoah, Historic...   
6                  3623                                                NaN   
7                 10613  Brickell, Central Business District (CBD), Dow...   
8                  6675                                                NaN   
9                  5665           Alameda, Grapeland Heights, and Fairlawn   
10                 4130                                                NaN   
11                 2148                                                NaN   
12                 3733                                                NaN   
13                 3840                      Lemon City (aka Little River)   
14                 8423  Riverside and South River Drive Historic District   
15                 3680                                                NaN   
16                    -                              Edgewater and Wynwood   
17                 3405                                      Spring Garden   
18                 3635                                                NaN   
19                 4899                                                NaN   
20                 2513  Bay Point Estates, Bayside District, Belle Mea...   
21                  NaN               Biscayne Island and San Marco Island   
22                    -                                           

__*NOTE:*__ Postal codes can be found at https://worldpostalcode.com/united-states/florida/miami . Some neighborhoods have more than one postal code but only one was recorded in the table above for simplicity purposes.

In [145]:
df_Mia.shape

(26, 7)

### Census data for average income, businesses, and retail sales

Obtain and read in data from the NY census to analyze in the report how population, businesses, and the economy of the city might influence the future of the clothing store. 

In [146]:
census_Mia= pd.read_html('https://www.census.gov/quickfacts/fact/table/miamicityflorida/POP060210')
census_Mia

[  All Topics Population Age and Sex Race and Hispanic Origin Population Characteristics Housing Families & Living Arrangements Computer and Internet Use Education Health Economy Transportation Income & Poverty Businesses Geography  \
 0                   Population per square mile, 2010                                                                                                                                                                                     
 
    Miami city, Florida  
 0              11135.9  ,
                                            Population Unnamed: 1
 0         Population estimates, July 1, 2019, (V2019)     467963
 1   Population estimates base, April 1, 2010, (V2019)     399481
 2   Population, percent change - April 1, 2010 (es...      17.1%
 3                   Population, Census, April 1, 2010     399457
 4                                         Age and Sex        NaN
 5                      Persons under 5 years, percent     5.8%
 6           

### Estimate population growth for Miami

In [154]:
pop2010_Mia= 399481
pop2019_Mia= 467963
pop_growth_Mia= ((pop2019_Mia-pop2010_Mia)/(pop2010_Mia))*100
print('The population growth rate in Miami (from 2010-2019) was around:', pop_growth_Mia, '%')

The population growth rate in Miami (from 2010-2019) was around: 17.142742708664493 %


In [156]:
futurepop_Mia= (pop2019_Mia*(pop_growth_Mia/100))+pop2019_Mia
print('The future estimated population in Miami in 9 years (assuming the growth rate remains the same) is: ', futurepop_Mia)

The future estimated population in Miami in 9 years (assuming the growth rate remains the same) is:  548184.6930617476


### Data for costs of living in Miami

In [157]:
livingMia= pd.read_html('https://www.numbeo.com/cost-of-living/in/Miami')
livingMia

[    0                                                  1
 0 NaN  whenDocReady(function() {  jQuery('#city_selec...,     0      1
 0 NaN  Tweet,                                           Restaurants     [ Edit ]  \
 0                        Meal, Inexpensive Restaurant      15.00 $   
 1   Meal for 2 People, Mid-range Restaurant, Three...      70.00 $   
 2      McMeal at McDonalds (or Equivalent Combo Meal)       8.00 $   
 3                      Domestic Beer (1 pint draught)       6.00 $   
 4                  Imported Beer (12 oz small bottle)       8.00 $   
 5                                Cappuccino (regular)       4.20 $   
 6                     Coke/Pepsi (12 oz small bottle)       2.07 $   
 7                          Water (12 oz small bottle)       1.73 $   
 8                                             Markets     [ Edit ]   
 9                          Milk (regular), (1 gallon)       3.71 $   
 10                   Loaf of Fresh White Bread (1 lb)       2.75 $   
 11 

### Create a map of Miami

In [368]:
latitudeMia= 25.7617
longitudeMia= -80.1918
map_Mia= folium.Map(location= [latitudeMia, longitudeMia], zoom_start= 12)
map_Mia

### Use Foursquare to find popular clothing venues in Miami

In [271]:
CLIENT_ID = '1IVZFEEO0D5XWSHQUVP0UIMMFSPSPDXWYFMCCFMFEMEAMLXJ' 
CLIENT_SECRET = 'LRRZSM13LBPVZ1PHJQX0PDZML2C1QJQM2HNQ2JEXSWAK4S2V' 
VERSION = '20180605'
LIMIT = 30

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 1IVZFEEO0D5XWSHQUVP0UIMMFSPSPDXWYFMCCFMFEMEAMLXJ
CLIENT_SECRET:LRRZSM13LBPVZ1PHJQX0PDZML2C1QJQM2HNQ2JEXSWAK4S2V


In [341]:
address2 = 'Miami, FL'
latitude2= 25.6905
longitude2= -80.3125
print(latitude2, longitude2)

25.6905 -80.3125


In [342]:
search_query2 = 'mall'
radius = 1000
print(' Searching for:',search_query +  '.... OK!')

 Searching for: clothing.... OK!


In [343]:
url2 = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&query={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude2, longitude2, VERSION, search_query2, radius, LIMIT)
url2

'https://api.foursquare.com/v2/venues/search?client_id=1IVZFEEO0D5XWSHQUVP0UIMMFSPSPDXWYFMCCFMFEMEAMLXJ&client_secret=LRRZSM13LBPVZ1PHJQX0PDZML2C1QJQM2HNQ2JEXSWAK4S2V&ll=25.6905,-80.3125&v=20180604&query=mall&radius=1000&limit=30'

In [344]:
results2 = requests.get(url2).json()
results2

{'meta': {'code': 200, 'requestId': '5f371cfa3694f37b7b31dbe0'},
 'response': {'venues': [{'id': '4b14ca5cf964a5208da623e3',
    'name': 'Dadeland Mall',
    'location': {'address': '7535 N Kendall Dr',
     'crossStreet': 'Dadeland Blvd',
     'lat': 25.69054534138613,
     'lng': -80.3124945912628,
     'labeledLatLngs': [{'label': 'display',
       'lat': 25.69054534138613,
       'lng': -80.3124945912628}],
     'distance': 5,
     'postalCode': '33156',
     'cc': 'US',
     'city': 'Miami',
     'state': 'FL',
     'country': 'United States',
     'formattedAddress': ['7535 N Kendall Dr (Dadeland Blvd)',
      'Miami, FL 33156',
      'United States']},
    'categories': [{'id': '4bf58dd8d48988d1fd941735',
      'name': 'Shopping Mall',
      'pluralName': 'Shopping Malls',
      'shortName': 'Mall',
      'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/shops/mall_',
       'suffix': '.png'},
      'primary': True}],
    'referralId': 'v-1597447855',
    'hasPerk': Fal

In [345]:
venues2 = results2['response']['venues']

# tranform venues into a dataframe
dataframe2 = json_normalize(venues2)
dataframe2

categories  hasPerk  \
0   [{'id': '4bf58dd8d48988d1fd941735', 'name': 'S...    False   
1   [{'id': '4bf58dd8d48988d1ce941735', 'name': 'S...    False   
2   [{'id': '4bf58dd8d48988d107951735', 'name': 'S...    False   
3   [{'id': '4bf58dd8d48988d1fd941735', 'name': 'S...    False   
4   [{'id': '4bf58dd8d48988d117951735', 'name': 'C...    False   
5   [{'id': '4bf58dd8d48988d1f2941735', 'name': 'S...    False   
6   [{'id': '4bf58dd8d48988d108951735', 'name': 'W...    False   
7   [{'id': '4bf58dd8d48988d122951735', 'name': 'E...    False   
8   [{'id': '4bf58dd8d48988d1fd941735', 'name': 'S...    False   
9   [{'id': '4bf58dd8d48988d171941735', 'name': 'E...    False   
10  [{'id': '4bf58dd8d48988d1ca941735', 'name': 'P...    False   
11  [{'id': '4bf58dd8d48988d104951735', 'name': 'B...    False   
12  [{'id': '4bf58dd8d48988d10c951735', 'name': 'C...    False   
13  [{'id': '4bf58dd8d48988d103951735', 'name': 'C...    False   
14  [{'id': '4bf58dd8d48988d102951735', 'name': 'A...    False   
15  [{'id': '4bf58dd8d48988d1f3941735', 'name': 'T...    False   
16                                                 []    False   
17  [{'id': '4bf58dd8d48988d11a941735', 'name': 'O...    False   
18  [{'id': '4f04afc02fb6e1c99f3db0bc', 'name': 'M...    False   
19  [{'id': '4bf58dd8d48988d16e941735', 'name': 'F...    False   
20  [{'id': '4bf58dd8d48988d103951735', 'name': 'C...    False   
21  [{'id': '4bf58dd8d48988d111951735', 'name': 'J...    False   
22  [{'id': '4bf58dd8d48988d108951735', 'name': 'W...    False   
23  [{'id': '4bf58dd8d48988d1f6941735', 'name': 'D...    False   
24  [{'id': '4bf58dd8d48988d103951735', 'name': 'C...    False   
25  [{'id': '4bf58dd8d48988d103951735', 'name': 'C...    False   
26  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...    False   
27  [{'id': '4bf58dd8d48988d16e941735', 'name': 'F...    False   
28  [{'id': '4bf58dd8d48988d107951735', 'name': 'S...    False   
29  [{'id': '4bf58dd8d48988d10c951735', 'name': 'C...    False   

                          id                location.address location.cc  \
0   4b14ca5cf964a5208da623e3               7535 N Kendall Dr          US   
1   52694f11498e980ab15d1af8               7535 N Kendall Dr          US   
2   568435fe498e546e328d3958              7535 N Kendall Dr.          US   
3   4bc47d42920eb71304c51e2c               7700 N Kendall Dr          US   
4   5e616b8b35ec690008a119a6        7535 N Kendall Dr # K148          US   
5   5332f34e498e79e651781035                             NaN          US   
6   59e2afc20d173f42885d36f3                       72 Avenue          US   
7   4e15019c52b1b9e5643ce942              7535 N Kendall Dr.          US   
8   4f1a2db07beb3367ac3b4580               7535 N Kendall Dr          US   
9   4fb6d86e6b74d33d6926eace               7535 N Kendall Dr          US   
10  4d0fb7cfe2365481157273ea                      SW 85th St          US   
11  5da5c4cb2d061800081b3bec        7597 North Kendall Drive          US   
12  5075ec8ce4b0e2c61e576aa6                             NaN          US   
13  5ae728c2c97f28002c023a56         7535 N Kendall Dr #2440          US   
14  4d324e51c6cba35d2b20207a         7681 N Kendall Dr #2240          US   
15  4fd12a97e4b0ce6eca84d22a                             NaN          US   
16  516575ccf1366a27311596bb               7441 N Kendall Dr          US   
17  4ec4bde37beb90d7e2c8efa2               7509 N Kendall Dr          US   
18  4ccc953cb571b60c45d8d365         7535 N Kendall Dr #K147          US   
19  4bcdceec0687ef3ba374e1cc               7537 N Kendall Dr          US   
20  4c424c57ce54e21e22810c1a  7535 Dadeland Mall, Space 1650          US   
21  4bddcde7587b2d7f89265509      7363 N Kendall Dr Spc 1690          US   
22  4c7a295fa8683704a1f5124d               7449 N Kendall Dr          US   
23  4aa6ca41f964a520fb4a20e3       7303 N KENDALL DR, 1st Fl          US   
24  4d9747410caaa14372a88db3                            US 1          US   
25  4b79dc9af964a520

In [346]:
filtered_columns2 = ['name', 'categories'] + [col for col in dataframe2.columns if col.startswith('location.')] + ['id']
dataframe_filtered2 = dataframe2.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered2['categories'] = dataframe_filtered2.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered2.columns = [column.split('.')[-1] for column in dataframe_filtered2.columns]

dataframe_filtered2

name            categories  \
0                                       Dadeland Mall         Shopping Mall   
1        SHRIMP HOUSE - Seafood Pasta - Dadeland Mall    Seafood Restaurant   
2               Donald J Pliner Miami - Dadeland Mall            Shoe Store   
3                                       Greenery Mall         Shopping Mall   
4                            Sugar King Dadeland Mall           Candy Store   
5                               fit2run dadeland mall   Sporting Goods Shop   
6                  New York & Company - Dadeland Mall         Women's Store   
7                           Cellairis - Dadeland Mall     Electronics Store   
8                   Soho Fashion Inc at Dadeland Mall         Shopping Mall   
9   Maroon 5 Payphone Simon Properties / Dadeland ...           Event Space   
10                 Cozzoli's Pizzeria - Dadeland Mall           Pizza Place   
11                            A|X Miami Dadeland Mall              Boutique   
12                      Bayside Brush - Dadeland Mall        Cosmetics Shop   
13                             Sportive Dadeland Mall        Clothing Store   
14                   Optical Elements - Dadeland Mall     Accessories Store   
15                     Kiddie Play Circle In The Mall      Toy / Game Store   
16                       Koko & Palenki Dadeland Mall                  None   
17                                       Hot Magazine       Other Nightlife   
18                                           T-Mobile     Mobile Phone Shop   
19                     Bannastrow's Crepes and Coffee  Fast Food Restaurant   
20                                             J.Crew        Clothing Store   
21                                          Swarovski         Jewelry Store   
22                           White House Black Market         Women's Store   
23                                             Macy's      Department Store   
24                                              Cache        Clothing Store   
25                                   Urban Outfitters        Clothing Store   
26                             The Cheesecake Factory   American Restaurant   
27                                        Chick-fil-A  Fast Food Restaurant   
28                                          Florsheim            Shoe Store   
29                                            SEPHORA        Cosmetics Shop   

                           address  cc             city        country  \
0                7535 N Kendall Dr  US            Miami  United States   
1                7535 N Kendall Dr  US          Kendall  United States   
2               7535 N Kendall Dr.  US            Miami  United States   
3                7700 N Kendall Dr  US            Miami  United States   
4         7535 N Kendall Dr # K148  US            Miami  United States   
5                              NaN  US          Kendall  United States   
6                        72 Avenue  US            Miami  United States   
7               7535 N Kendall Dr.  US            Miami  United States   
8                7535 N Kendall Dr  US            Miami  United States   
9                7535 N Kendall Dr  US            Miami  United States   
10                      SW 85th St  US  Kendall-Perrine  United States   
11        7597 North Kendall Drive  US            Miami  United States   
12                             NaN  US            Miami  United States   
13         7535 N Kendall Dr #2440  US            Miami  United States   
14         7681 N Kendall Dr #2240  US            Miami  United States   
15                             NaN  US            Miami  United States   
16               7441 N Kendall Dr  US            Miami  United States   
17               7509 N Kendall Dr  US            Miami  United States   
18         7535 N Kendall Dr #K147  US            Miami  United States   
19               7537 N Kendall Dr  US            Miami  United States   
20  7535 Dadeland Mall, Space 1650  US       

__*NOTE:*__: I specifically searched for the area closest to Dadeland Mall to avoid overcrowding of data.

In [363]:
lat= dataframe2['location.lat']
lng= dataframe2['location.lng']
dataframe_filtered2['lat']= lat
dataframe_filtered2['lng']= lng

/opt/conda/envs/Python36/lib/python3.6/site-packages/ipykernel/__main__.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  app.launch_new_instance()
/opt/conda/envs/Python36/lib/python3.6/site-packages/ipykernel/__main__.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [367]:
venues2_map = folium.Map(location=[latitude2, longitude2], zoom_start=15)

for lat, lng, label in zip(dataframe_filtered2.lat, dataframe_filtered2.lng, dataframe_filtered2.name):
    folium.features.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues2_map)

# display map
venues2_map